### PR TITLE
SearchView clears on backPressed

### PR DESCRIPTION
**Pull request by nitish1211**
Fix #269 

Sorry had to create a new pull request. But I believe this won't create a problem while merging.

@Sam1301 Can you please take a look at this.

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@nitish1211, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/).

**Comment by Sam1301** [comment]
For git related issues, we have a beginner friendly guide http://zulip.readthedocs.io/en/latest/git-guide.html :) For more, you can also checkout this tutorial https://www.atlassian.com/git/tutorials/what-is-version-control.

**Comment by nitish1211** [comment]
@Sam1301 The searchview clears on another backpress after coming back to home screen.

As you can see in this.

g_20161228_0427014

**Comment by Sam1301** [comment]
@nitish1211 Oh I see, you have accounted for 2 back presses before removing searchview: 1) to pull down the keyboard 2) to go back to chat screen. Consider a case where the user presses the up navigation button to move back to chat screen, then considering this logic, the user would have to make one extra back press which wouldn't cause any UI interaction. 
How about we pull the keyboard down when search is started and clear searchview on a single back press or up navigation?

**Comment by kunall17** [comment]
@Sam1301 I would agree on that! 

> (clear searchview on a single back press or up navigation)

@nitish1211 Can you please 
- Squash the commits (https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/)
- It would be nice to display a back icon replacing the drawer icon when search is initiated
- I see this is your new PR for this feature (old #269) , please don't create new PR's rather use rebase (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Changing-the-Last-Commit) from the next time

**Comment by nitish1211** [comment]
@Sam1301 @kunall17 
 I have added the features you requested : 
1)The Keyboard hides on searching
2)Pressing of home button take you back in one press
3)There are no extra back presses required to go back to the home screen after searching.

g_20170103_0129254

**Comment by nitish1211** [comment]
And also Squashed :)

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 1]
```diff
@@ -1457,16 +1459,24 @@ public void run() {
               narrowedList = null;
               getSupportFragmentManager().popBackStack(NARROW,
                     FragmentManager.POP_BACK_STACK_INCLUSIVE);
+            clearSearch();
         }
     }
 
+    public void clearSearch()
+    {
+        searchView.setIconified(true);
```
Comment (Sam1301) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Nice! Here maybe a comment is in need to explain that the first call sets empty string in searchview textview and second call hides keyboard. Also, it be helpful if you can use JavaDoc comments for newly introduced functions. Once `.travis.yml` is merged we may set the severity of missing JavaDoc comments to error just to enhance code readability.

**Comment by Sam1301** [comment]
Nice! Thanks for working on this @nitish1211... the code looks fine to me :+1:

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 2]
```diff
@@ -1434,9 +1435,10 @@ public void clearChatBox() {
         }
     }
 
+    @Override
     public void onBackPressed() {
         if (narrowedList == null) {
-
+
```
Comment (timabbott) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Is this whitespace change intended?

**Comment by timabbott** [comment]
@nitish1211 can you sign the Dropbox CLA?

**Comment by timabbott** [comment]
Also, it'd be great if you cleaned up the commit message a bit to follow Zulip's style guidelines.  
Check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages.

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java (hunk 2 above)
Comment (nitish1211) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Oh sorry about that. I'll fix that.

**Comment by kunall17** [comment]
@nitish1211 you missed this one

> It would be nice to display a back icon replacing the drawer icon when search is initiated

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*
				
@nitish1211, thanks for signing the CLA!

**Comment by nitish1211** [comment]
@kunall17 I've added the back button on searching too.

g_20170104_1727403

**Comment by Sam1301** [comment]
@nitish1211 It'd be nice if you could clean up the commit message. Just run `git commit --amend` to edit the last commit. I think its ok to just include `Fixes #269` as the commit description. Check out the Zulip commit message guidelines for more details: https://zulip.readthedocs.io/en/latest/version-control.html#commit-messages.
Also, to see if your commit will pass the checks, you can run `./gradlew lintDebug` and `./gradlew assembleDebug` in your terminal.

**Comment by Sam1301** [comment]
Also, it looks like some old commits are also part of your pr.

**Pull request closed without merging**